File: client/src/main/js/components/reports/BuildReport.js
import React from 'react';

export function BuildReport(props) {
    const latestBuildReport = props.solution.buildReports.last();

    if (latestBuildReport) {
        if (latestBuildReport.succeed) {
            return <section className="report successful-build-report"><i className="material-icons">done</i></section>;
        } else {
            return <section className="report failed-build-report"><i className="material-icons">clear</i></section>;
        }
    } else {
        return <section className="report"/>
    }
}

File: frontend/src/main/kotlin/org/flaxo/frontend/component/report/BuildReport.kt
package org.flaxo.frontend.component.report

import org.flaxo.frontend.data.Solution
import react.RBuilder
import react.dom.i

fun RBuilder.buildReport(solution: Solution) = solution.buildReports
        .lastOrNull()
        ?.also {
            if (it.succeed) {
                i(classes = "material-icons successful-build-report") { +"done" }
            } else {
                i(classes = "material-icons failed-build-report") { +"clear" }
            }
        }
        ?: i(classes = "material-icons") { }